File: backend/app/routers/ai.py
from fastapi import APIRouter, HTTPException

from ..services.ai_service import ai_service
from ..schemas import AiRewriteRequest

router = APIRouter(prefix="/api/ai", tags=["ai"])


@router.post("/rewrite-text")
async def rewrite_text(request: AiRewriteRequest):
    try:
        result = await ai_service.rewrite_bullet(request.text, request.target_position)
        return {"result": result}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/diagnose")
async def diagnose_resume(resume_data: dict):
    try:
        result = await ai_service.diagnose_resume(resume_data)
        return {"result": result}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/jd-match")
async def jd_match(resume_data: dict, jd_text: str):
    try:
        result = await ai_service.jd_match(resume_data, jd_text)
        return {"result": result}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/optimize-section")
async def optimize_section(section: str, content: str, target_position: str = ""):
    try:
        result = await ai_service.optimize_section(section, content, target_position)
        return {"result": result}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: frontend/src/api/ai.ts
import axios from 'axios'

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL || 'http://localhost:8000'

const api = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
})

export interface AiRewriteRequest {
  text: string
  target_position?: string
}

export interface AiRewriteResponse {
  result: string
}

export async function rewriteText(request: AiRewriteRequest): Promise<AiRewriteResponse> {
  const response = await api.post('/api/ai/rewrite-text', request)
  return response.data
}

export async function diagnoseResume(resumeData: Record<string, any>): Promise<AiRewriteResponse> {
  const response = await api.post('/api/ai/diagnose', resumeData)
  return response.data
}

export async function jdMatch(resumeData: Record<string, any>, jdText: string): Promise<AiRewriteResponse> {
  const response = await api.post('/api/ai/jd-match', { resume_data: resumeData, jd_text: jdText })
  return response.data
}

export async function optimizeSection(section: string, content: string, targetPosition?: string): Promise<AiRewriteResponse> {
  const response = await api.post('/api/ai/optimize-section', { section, content, target_position: targetPosition })
  return response.data
}